src/Containers/SparseAxisArray.jl

```diff
@@ -275,27 +275,26 @@ function Base.show(io::IO, ::MIME"text/plain", sa::SparseAxisArray)
     end
 end
 Base.show(io::IO, x::SparseAxisArray) = show(convert(IOContext, io), x)
+
 function Base.show(io::IOContext, x::SparseAxisArray)
-    # TODO: make this a one-line form
     if isempty(x)
         return show(io, MIME("text/plain"), x)
     end
-    limit::Bool = get(io, :limit, false)
+    limit = get(io, :limit, false)::Bool
     half_screen_rows = limit ? div(displaysize(io)[1] - 8, 2) : typemax(Int)
-    key_string(key::Tuple) = join(key, ", ")
     print_entry(i) = i < half_screen_rows || i > length(x) - half_screen_rows
+    # For stable printing, sort by the string value of the key.
+    key_strings = [(join(key, ", "), value) for (key, value) in x.data]
+    sort!(key_strings; by = x -> x[1])
```

Comment:
What  if the key does not implement `isless` ? Probably using an `OrderedDict` is easier

Reply:
It does it on the string representation of the key, which we were already producing. This seemed much easier than an OrderedDict.

Reply:
Indeed, it should work then. I'm just thinking that if we are using an ordered dict for affine expression to make sure things are always easily reproducible then we might as well do it for `SparseAxisArray`. Or we need to convince ourself that it will never affect any behaviors of JuMP.

Reply:
That requires a bigger change to how SparseAxisArray is constructed. This seemed like an easy win in the near-term.

Reply:
Ok, let's keep that in mind